daft/context.py

```diff
@@ -161,21 +165,39 @@ def daft_planning_config(self) -> PyDaftPlanningConfig:
             return self._daft_planning_config
 
     @property
-    def runner_config(self) -> _RunnerConfig:
+    def runner_config(self) -> _RunnerConfig | None:
+        """Retrieve the currently active runner config, or None if it has not yet been set"""
+        with self._lock:
+            return self._runner_config
+
+    def get_runner_config_name(self) -> Literal["ray"] | Literal["py"] | Literal["native"]:
         with self._lock:
-            return self._get_runner_config()
+            if self._runner_config is None:
+                return _get_runner_config_from_env().name
+            else:
+                return self._runner_config.name
```

Comment:
Maybe instead of this, we can perform checks in our tests against `os.getenv("DAFT_RUNNER")`